core/src/main/scala/cats/MonadReader.scala

```diff
@@ -0,0 +1,14 @@
+package cats
+
+/** A monad that has the ability to read from an environment. */
+trait MonadReader[F[_, _], R] extends Monad[F[R, ?]] {
+  /** Get the environment */
+  def ask: F[R, R]
+
+  /** Modify the environment */
+  def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

Comment:
Ah your point about the flow is a good one. I didn't think about that.

One reason I like about putting the function in the last parameter list is that I often am using a lambda that I format as multiple lines:

```scala
(...).local(fra)(r =>
  doStuffWithR(r))
```

Which looks slightly nicer to me than:

```scala
(...).local(r =>
  doStuffWithR(r)
)(fra)
```

So between the convention on other methods, and this formatting, I'd be a bit inclined to have the function in a second parameter list; but it's not a big deal to me. Whichever people prefer.